src/Appwrite/Platform/Tasks/Backup.php

```diff
@@ -0,0 +1,272 @@
+        self::log($stderr);
+
+        if (!str_contains($stderr, 'completed OK!') || !file_exists($target . '/xtrabackup_checkpoints')) {
+            Console::error('Backup failed');
+            Console::exit();
+        }
+
+        if (!unlink($logfile)) {
+            Console::error('Error deleting: ' . $logfile);
+            Console::exit();
+        }
+    }
+
+    public function tar(string $directory, string $file)
+    {
+        $stdout = '';
+        $stderr = '';
+
+        $cmd = 'cd ' . $directory . ' && tar zcf ' . $file . ' . && cd ' . getcwd();
+        self::log($cmd);
+        Console::execute($cmd, '', $stdout, $stderr);
+        self::log($stdout);
+        if (!empty($stderr)) {
+            Console::error($stderr);
+            Console::exit();
+        }
+
+        if (!file_exists($file)) {
+            Console::error('Can\'t find tar file: ' . $file);
+            Console::exit();
+        }
+
+        $filesize = \filesize($file);
+        self::log('Tar file size is: ' . ceil($filesize / 1024 / 1024) . 'MB');
+        if ($filesize < (2 * 1024 * 1024)) {
+            Console::error('File size is very small: ' . $file);
+            Console::exit();
+        }
+    }
+
+    public function upload(string $file, Device $local)
+    {
+        $filename = basename($file);
+
+        if (!$this->s3->exists('/')) {
+            Console::error('Can\'t read s3 root directory');
+            Console::exit();
+        }
+
+        try {
+            self::log('Uploading: ' . $file);
+            $destination = $this->s3->getRoot() . '/' . $filename;
+
+            if (!$local->transfer($file, $destination, $this->s3)) {
+                Console::error('Error uploading to ' . $destination);
+                Console::exit();
+            }
+
+            if (!$this->s3->exists($destination)) {
+                Console::error('File not found in destination: ' . $destination);
+                Console::exit();
+            }
+        } catch (Exception $e) {
+            Console::error($e->getMessage());
+            Console::exit();
+        }
+
+        if (!unlink($file)) {
+            Console::error('Error deleting: ' . $file);
+            Console::exit();
+        }
+    }
+
+    public static function log(string $message): void
+    {
+        if (!empty($message)) {
+            Console::log(date('Y-m-d H:i:s') . ' ' . $message);
+        }
+    }
+
+    public function checkEnvVariables(): void
+    {
+        foreach (
+            [
+                '_APP_CONNECTIONS_BACKUPS_STORAGE',
+                '_APP_CONNECTIONS_DB_REPLICAS',
+            ] as $env
+        ) {
+            if (empty(App::getEnv($env))) {
+                Console::error('Can\'t read ' . $env);
+                Console::exit();
+            }
+        }
+    }
+
+    public function getDsn(string $database): ?DSN
+    {
+        foreach (explode(',', App::getEnv('_APP_CONNECTIONS_DB_REPLICAS')) as $project) {
```

Comment:
Let's add a default for `App::getEnv`, even if it's just an empty string. The current parameter default is null, and passing null to explode is deprecated

Reply:
Cool, Added default parameter 👍